### DISCUSSION TITLE
[BUG] Release event "prereleased" not firing/working if the release was previously a draft

### DISCUSSION THREAD

**Discussion by malimccalla** [community/community#26954] [Actions]
<p><strong>Description</strong></p>
<p>On github If I create a draft version of a prelease and the go back to publish it later it is not picked up by my workflow.</p>
<pre><code>name: Prelease bug

on:
  release:
    types: [prereleased]

jobs:
  job_1:
    name: Test
    runs-on: ubuntu-latest
    steps:
      - run: echo "hello"
</code></pre>
<p><strong>Steps to reproduce:<br><br></strong>1. Go to <a href="https://github.com/malimccalla/simple-k8s/releases/new" rel="nofollow noopener">https://github.com/&lt;username&gt;/&lt;repo&gt;/releases/new</a><br>
2. Fill out release details and tick the “This is a pre-release” checkbox<br>
3. Click “Save draft”<br>
4. Navigate to the releases tab on gibhub to see your drafted release and click “edit”<br>
5. Click “Publish release”</p>
<p><strong>What happens:<br><br></strong>My workflow does not run</p>
<p><strong>What I expect to happen:<br><br></strong>My workflow to run</p>
<p><strong>Additional information:<br><br></strong>This happens when the “This is a pre-release” checkbox is ticked <strong><em>before</em></strong> a draft is saved. If I create a draft without ticking the box and then come back to the draft to publish and tick the box then all is fine and my workflow runs </p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/malimccalla">@malimccalla</a> ,</p>
<p>Thank you for reaching this out!  This is by designed. As <a href="https://help.github.com/en/actions/reference/events-that-trigger-workflows#release-event-release" rel="nofollow noopener">doc</a> mentioned:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/b/bb884596ca48aff30daf9cb603cac8583678e4e1.png" data-download-href="/uploads/short-url/qKZfnpXZcgqcjyIpOUb41NvMI2B.png?dl=1" title="releasedraft.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181100940-97f45ad7-66c8-46c9-bb5d-8ef73fb93e74.png" alt="releasedraft.png" data-base62-sha1="qKZfnpXZcgqcjyIpOUb41NvMI2B" width="690" height="86" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/b/bb884596ca48aff30daf9cb603cac8583678e4e1_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">releasedraft.png</span><span class="informations">770×97 4.13 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Hence, for your step3, no workflow is triggered for draft release.</p>
<p>For your step 5, no operation to tick the ‘prereleased’ checkbox, the action type is actually ‘published’.</p>
<p>I add all activity event type in yaml file and confirm in github context:</p>
<p><img src="https://user-images.githubusercontent.com/52772761/181100941-9b53076c-13bd-4fe1-8c04-1545da553987.png" alt="publishaction.png" data-base62-sha1="41aCAhun4TCucJ2r02ST0m5xBti" width="280" height="76"></p>
<p>For your additional information test, the action type can be ‘prereleased’.</p>
<p><img src="https://user-images.githubusercontent.com/52772761/181100942-9adade59-395f-4d74-80ff-b32a6d6fc35b.png" alt="preleased.png" data-base62-sha1="ncVk4oytmVs0FwkNovRCDs7ZwZj" width="280" height="81"></p>
<p>It’s recommened to use <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#example-printing-context-information-to-the-log-file" rel="nofollow noopener">github context</a> to check the detail action type.</p>
<p>Hope it clear and helpful!</p>

**Comment by malimccalla** [comment]
<p><a class="mention" href="/weide-zhou">@weide-zhou</a> Thanks for clearing that up for me. I was just a little confused as the release <code>released</code> event does still fire if the drafted release is not a prerelease. Thanks for your help.</p>
